client/src/Components/nav.js

```diff
@@ -0,0 +1,174 @@
+import {
+  faBars,
+  faChevronLeft,
+  faFutbol,
+  faHeart,
+  faSearch,
+  faShoppingCart,
+  faSignInAlt,
+} from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useEffect, useRef, useState } from "react";
+import { NavLink, useNavigate } from "react-router-dom";
+import { useAuth } from "../Auth/auth-context";
+import { useCart } from "../cart-context/cart-context";
+
+export const Navigation = () => {
+  const [searchText, setSearchText] = useState("");
+  const [showSearchBox, setShowSearchBox] = useState(false);
+  const { state, logoutFunction } = useAuth();
+  const {
+    cartDispatch,
+    navbar,
+    cartState: { cartItems, wishList },
+  } = useCart();
+  const [showOptions, setShowOptions] = useState(false);
+  const menuBtn = useRef(null);
+  const navigate = useNavigate();
+
+  const wishlist_count = wishList.length ? wishList.length : null;
+  const cart_count = cartItems.length ? cartItems.length : null;
+  useEffect(() => {
+    document.addEventListener("click", (e) => {
```

Comment:
not sure what this piece of code is doing......we can discuss that?

Reply:
yeah definitely..it's handling the navbar